### PR TITLE
Migrate MontySolr to Gradle

### PR DESCRIPTION
**Pull request by JCRPaquin**
Replaces Ant with Gradle to lower build times and provide better support for common developer tools.

### CODE REVIEW

**Comment by JCRPaquin** [comment]
Can't be merged yet because tests aren't passing, but I'd like to solicit comments from folks early given the size of this change.

File: montysolr/src/test/java/monty/solr/util/SolrTestSetup.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+        // Add repo root + ADS solr conf to classpath
+        Path solrConfPath = Paths.get("deploy", "adsabs", "server", "solr", "collection1", "conf");
+        URLClassLoader extendedLoader = URLClassLoader.newInstance(
+                new URL[] {getRepoUrl(), getRepoUrl(solrConfPath)},
+                SolrTestSetup.class.getClassLoader());
+
+        SolrResourceLoader loader = new SolrResourceLoader(solrPath,
+                extendedLoader);
+        SolrConfig solrConfig = SolrConfig.readFromResourceLoader(loader, config);
```
Comment (shinyichen) [inline montysolr/src/test/java/monty/solr/util/SolrTestSetup.java:38]: Is this no longer used?

File: montysolr/src/test/java/org/apache/solr/analysis/TestAdsabsTypeDateString.java [hunk 2]
```diff
@@ -275,7 +275,7 @@ public void test() throws Exception {
          */
 
         for (int i = 1900; i < 2025; i++) {
-            assertQ(Integer.toString(i),
+            assertQ(req("q", Integer.toString(i)),
```
Comment (shinyichen) [inline montysolr/src/test/java/org/apache/solr/analysis/TestAdsabsTypeDateString.java:271]: Not sure if this change is correct. By looking at the rest of the test, my understanding is `q` is the query param of ADS search, and `q=1900` is not a valid query. If we are querying a single year, it should be `q=year:1900`. Maybe this line is suppose to be 
`assertQ(req("q", "year=" + Integer.toString(i)), "//*[@numFound='1']")`

File: montysolr/src/test/java/monty/solr/util/SolrTestSetup.java (hunk 1 above)
Comment (JCRPaquin) [reply montysolr/src/test/java/monty/solr/util/SolrTestSetup.java:38]: It doesn't seem to be, you're right. There don't appear to be any side-effects caused by reading in the config and the tests run fine. I think this was bypassed when I started creating the test harness manually.

File: montysolr/src/test/java/org/apache/solr/analysis/TestAdsabsTypeDateString.java (hunk 2 above)
Comment (JCRPaquin) [reply montysolr/src/test/java/org/apache/solr/analysis/TestAdsabsTypeDateString.java:271]: I originally kept this query as just `Integer.toString(i)` to maintain the spirit of the original test. It also works with `req("q", "year:" + i)` and I don't have a problem switching to that. Just wasn't sure whether we wanted to support a bare year, which is what I thought this test implied.

Comment (shinyichen) [reply montysolr/src/test/java/org/apache/solr/analysis/TestAdsabsTypeDateString.java:271]: Not sure what this test was meant to do, I am only guessing that it's trying 'query by the year'

Comment (JCRPaquin) [reply montysolr/src/test/java/org/apache/solr/analysis/TestAdsabsTypeDateString.java:271]: Spoke with @kelockhart offline and we agreed that we should test both fielded and unfielded year queries here.

I've added a fielded query to this loop to ensure both the unfielded & fielded variants pass.

**Pull request merged**